pilot/pkg/serviceregistry/kube/controller/endpointslice.go

```diff
@@ -244,29 +245,55 @@ func endpointHealthStatus(svc *model.Service, e v1.Endpoint) model.HealthStatus
 	return model.UnHealthy
 }
 
-func (esc *endpointSliceController) updateEndpointCacheForSlice(hostName host.Name, slice *v1.EndpointSlice) {
+func (esc *endpointSliceController) updateEndpointCacheForSlice(hostName host.Name, epSlice *v1.EndpointSlice) {
 	var endpoints []*model.IstioEndpoint
-	if slice.AddressType == v1.AddressTypeFQDN {
+	if epSlice.AddressType == v1.AddressTypeFQDN {
 		// TODO(https://github.com/istio/istio/issues/34995) support FQDN endpointslice
 		return
 	}
 	svc := esc.c.GetService(hostName)
-	svcNamespacedName := getServiceNamespacedName(slice)
+	svcNamespacedName := getServiceNamespacedName(epSlice)
+	// This is not a endpointslice for service, ignore
```

Comment:
what if it used to have the label? Is this fixing a bug?

Reply:
We cannot remove the label, k8s will add it back

Yes, i am fixing to ignore no svc endpointslice.